### PR TITLE
Added eksa metal and workshop-project modules

### PR DESCRIPTION
**Pull request by dcallao**
This PR contains the following changes:
- added metal, eksa, and root modules
- added examples on root module
- produced genetic README.md with `terraform-docs` util

### CODE REVIEW

File: users.csv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+email,metro,plan
+danielcallao@gmail.com,sv,t3.small.x86
```
Comment (displague) [inline users.csv:2]: 👋

File: variables.tf [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+# Variables for "eksa" module
+variable "enable_eksa" {
+  type        = bool
+  description = "Enable EKS-A module"
+  default     = false
+}
+
+variable "eksa_cluster_name" {
```
Comment (displague) [inline variables.tf:81]: These variables should follow the addons pattern and use something like an object called ekas_config (I didn't check the naming convention).

Comment (displague) [reply variables.tf:81]: Same comment about metal_* variables in this file.

**Comment by dcallao** [comment]
@displague I have made changes to resemble the k8s addons format. Have a look when you have a moment

**Comment by dcallao** [comment]
@displague This is ready for your review! Thanks in advance.

**Pull request merged**